### PR TITLE
Build - Refactor + CircleCI Mascara builds

### PR DESCRIPTION
**Pull request by kumavis**

### CODE REVIEW

**Comment by kumavis** [comment]
```
npm i -g web-server
```
```
npx gulp build
web-server -r dist/mascara
```

**Comment by kumavis** [comment]
hello

**Comment by kumavis** [comment]
thanks @metamaskbot 👌

**Comment by kumavis** [comment]
forced push to squash like ~20 commits of messing with circleci config

**Comment by kumavis** [comment]
ok im pretty happy with this

File: .circleci/config.yml [hunk 1]
```diff
@@ -98,6 +98,32 @@ jobs:
...
             - dist
+      - store_artifacts:
+          path: dist/mascara
+          destination: builds/mascara
+      - store_artifacts:
+          path: builds
+          destination: builds
+      - run:
+          name: build:announce
+          command: |
```
Comment (frankiebee) [inline .circleci/config.yml]: this just seems like it should go in a script?

File: gulpfile.js [hunk 2]
```diff
@@ -41,65 +54,67 @@ gulp.task('dev:reload', function() {
...
+
+// copy mascara
+
+createCopyTasks('html:mascara', {
+  source: './mascara/',
+  pattern: 'proxy/index.html',
+  destinations: [`./dist/mascara/`],
+})
+
+function createCopyTasks(label, opts) {
```
Comment (frankiebee) [inline gulpfile.js]: space in between please! `createCopyTasks (`

Comment (frankiebee) [reply gulpfile.js]: it makes it a lot harder to just find the function definition

File: ui/app/css/itcss/settings/typography.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-@import url('/fonts/Font_Awesome/font-awesome.min.css');
+@import url('./fonts/Font_Awesome/font-awesome.min.css');
```
Comment (frankiebee) [inline ui/app/css/itcss/settings/typography.scss]: youve changed it from root to here? i'm not sure that is correct.

**Review by frankiebee — DISMISSED** [review]
really the only thing i care about is keeping with linting patterns so that the code is predictable not sure about the font stuff and the yml stuff is more of a preference then a request

File: ui/app/css/itcss/settings/typography.scss (hunk 3 above)
Comment (frankiebee) [reply ui/app/css/itcss/settings/typography.scss]: seems fine

File: .circleci/config.yml (hunk 1 above)
Comment (kumavis) [reply .circleci/config.yml]: that sounds reasonable

**Comment by frankiebee** [comment]
does not seem to be watching the scripts anymore

**Comment by tmashuang** [comment]
This will increase our build process by ~2-3 mins. Did a side-by-side comparison here:
screen shot 2018-03-29 at 5 54 23 pm

When developing in `gulp dev`, my changes do not seem to register. It seems to be changing the opera/dist.

**Comment by kumavis** [comment]
main reason the increase is that mascara and extension both build

**Comment by kumavis** [comment]
moar refactor with moar QA

**Review by tmashuang — APPROVED** [review]
LGTM

**Pull request merged**